agent/src/main/java/com/teamscale/jacoco/agent/Agent.java

```diff
@@ -71,42 +69,64 @@ public Agent(AgentOptions options,
 	protected void initServerEndpoints(Service spark) {
 		spark.get("/partition", (request, response) ->
 				Optional.ofNullable(options.getTeamscaleServerOptions().partition).orElse(""));
+		spark.get("/message", (request, response) ->
+				Optional.ofNullable(options.getTeamscaleServerOptions().message).orElse(""));
 		spark.post("/dump", this::handleDump);
 		spark.post("/reset", this::handleReset);
-		spark.post("/partition/" + PARTITION_PARAMETER, this::handleSetPartition);
+		spark.put("/partition", this::handleSetPartition);
+		spark.put("/message", this::handleSetMessage);
 	}
 
 	/** Handles dumping a XML coverage report for coverage collected until now. */
 	private String handleDump(Request request, Response response) {
 		logger.debug("Dumping report triggered via HTTP request");
 		dumpReport();
-		response.status(204);
+		response.status(HttpServletResponse.SC_NO_CONTENT);
 		return "";
 	}
 
 	/** Handles resetting of coverage. */
 	private String handleReset(Request request, Response response) {
 		logger.debug("Resetting coverage triggered via HTTP request");
 		controller.reset();
-		response.status(204);
+		response.status(HttpServletResponse.SC_NO_CONTENT);
 		return "";
 	}
 
 	/** Handles setting the partition name. */
 	private String handleSetPartition(Request request, Response response) {
-		String partition = request.params(PARTITION_PARAMETER);
+		String partition = request.body();
```

Comment:
Is the `charset=UTF-8` requirement actually necessary or does Spark does any magic on `Request.body()`? In other words, Spark needs to pick some encoding to convert from bytes to `String` and I wonder how it does this here. Can you please check.

Reply:
It reads the charset from the request, so UTF-8 should not be required. Removed it from the message.